李新强

wilbur-lxq opened the pull request:
学生李新强完成A2ME（Assets to marketcap）AT（Total Assets）Beta（CAPM Beta）LME（Size）LTurnover（Turnover）MktBeta（Market Beta）的因子练习

Commit:
Update main.py with new logic

weiguanwang commented on main.py:
```diff
@@ -2194,6 +2196,116 @@ def calc_tes(self, change_qnum=4, std_qnum=8, std_min_qnum=6):
 
         return data
 
+    #A2ME（Assets to marketcap）##这个因子好像在1555-1561行已经出现过了
```
好的，我看到了，谢谢。那把这个函数删掉吧。

weiguanwang commented on main.py:
```diff
@@ -2194,6 +2196,116 @@ def calc_tes(self, change_qnum=4, std_qnum=8, std_min_qnum=6):
+        return total_assets
+
+    #LME（Size）
+    def calc_lme(self):
```
请帮我比较一下，这个lme算出来的数字，和212行的calc_size是不是只是相差一个取log

wilbur-lxq replied:
是的是的😂我已经删除了lme函数

Commit:
更新了Rel2High Resid_Var ST_Rev C 股票因子的代码，
发现Variance因子在459-466行出现过 LT_Rev因子在968-976行出现过
发现待制作因子Spread（Bid-ask spread）在数据中没找到bid price 和 ask price的数据

weiguanwang commented on main.py:
```diff
@@ -2194,6 +2196,169 @@ def calc_tes(self, change_qnum=4, std_qnum=8, std_min_qnum=6):
+
+    #AT（Total Assets）
+    def calc_at(self):
+        total_assets = self.get_data('FS_Combas', 'A001000000')
```
这里需要将total asset 进行shift 3，和1272行一致

weiguanwang commented on main.py:
```diff
@@ -2194,6 +2196,169 @@ def calc_tes(self, change_qnum=4, std_qnum=8, std_min_qnum=6):
+            market_r = market_r.loc[combined_index]
+
+            corr = stock_r.corr(market_r) #计算当前这只股票的超额收益率（stock_r）和市场超额收益率（market_r）之间的相关性（皮尔逊相关系数）
+            std_stock = np.log1p(stock_r).rolling(min_obs_vol).std().iloc[-1]   # 计算当前这只股票的超额收益率（stock_r）在过去120天内的标准差；np.log1p(stock_r)取对数
```
标准差计算使用的对数收益率，这并不是收益率取对数的意思。对数收益率指的是log(今天价格/前一天价格），普通收益率是（今天价格-前一天价格）/前一天价格。在一天的时间长度上，对数收益率和普通收益率很接近。可以直接使用普通收益率计算标准差
